include/eosio/to_bin.hpp

```diff
@@ -13,210 +13,179 @@
 namespace eosio {
 
 template <typename S>
-result<void> to_bin(std::string_view sv, S& stream);
+bool to_bin(std::string_view sv, S& stream, std::string_view&);
 
 template <typename S>
-result<void> to_bin(const std::string& s, S& stream);
+bool to_bin(const std::string& s, S& stream, std::string_view&);
 
 template <typename T, typename S>
-result<void> to_bin(const std::vector<T>& obj, S& stream);
+bool to_bin(const std::vector<T>& obj, S& stream, std::string_view&);
 
 template <typename T, typename S>
-result<void> to_bin(const std::optional<T>& obj, S& stream);
+bool to_bin(const std::optional<T>& obj, S& stream, std::string_view&);
 
 template <typename... Ts, typename S>
-result<void> to_bin(const std::variant<Ts...>& obj, S& stream);
+bool to_bin(const std::variant<Ts...>& obj, S& stream, std::string_view&);
 
 template <typename... Ts, typename S>
-result<void> to_bin(const std::tuple<Ts...>& obj, S& stream);
+bool to_bin(const std::tuple<Ts...>& obj, S& stream, std::string_view&);
 
 template <typename T, typename S>
-result<void> to_bin(const T& obj, S& stream);
+bool to_bin(const T& obj, S& stream, std::string_view&);
 
 template <typename S>
-result<void> varuint32_to_bin(uint64_t val, S& stream) {
-   if (val >> 32)
-      return stream_error::varuint_too_big;
+void varuint32_to_bin(uint64_t val, S& stream) {
+   check( !(val >> 32), convert_stream_error( stream_error::varuint_too_big) );
    do {
       uint8_t b = val & 0x7f;
       val >>= 7;
       b |= ((val > 0) << 7);
-      auto r = stream.write(b);
-      if (!r)
-         return r.error();
+      stream.write(b);
    } while (val);
-   return outcome::success();
 }
 
 // !!! temp
+// TODO fix this to properly propagate the error
 inline void push_varuint32(std::vector<char>& bin, uint32_t v) {
    vector_stream st{ bin };
-   varuint32_to_bin(v, st).has_value();
+   varuint32_to_bin(v, st);
```

Comment:
`has_value()` was just returned true/false which was ignored. Seems like there is no TODO here.